internal/pkg/http/client/request.go

```diff
@@ -165,15 +165,18 @@ func (r *Request) SetContext(ctx context.Context) *Request {
 	return r
 }
 
-// WaitFor ensures that all remaining listeners will be deferred until subRequest done
+// WaitFor ensures that all listeners which:
+// - have not yet been executed, before calling WaitFor
+// - and were defined before calling WaitFor
+// will be deferred until subRequest will be done.
 // See TestWaitForSubRequest test.
 func (r *Request) WaitFor(subRequest *Request) {
-	// Error from the targetRequest will be set to the parent request, if it is already processed.
-	inheritError := r.IsDone()
-
 	r.lock.Lock()
 	defer r.lock.Unlock()
 
+	// Error from the targetRequest will be set to the parent request, if it is already processed.
+	inheritError := r.done
+
```

Comment:
s tymto nebol problem, ... ale pre istotu .. `IsDone` ma v sebe tiez zamok ... potom sa odomkol, a potom sa znova zamkol.
takto je to teraz pod jednym zamkom.